public/app/plugins/panel/canvas/InlineEdit.tsx

```diff
@@ -0,0 +1,199 @@
+import { cx, css } from '@emotion/css';
+import { get as lodashGet } from 'lodash';
+import React, { SyntheticEvent, useRef, useState, useMemo } from 'react';
+import Draggable from 'react-draggable';
+import { Resizable, ResizeCallbackData } from 'react-resizable';
+import { useClickAway } from 'react-use';
+
+import {
+  GrafanaTheme,
+  PanelModel,
+  Dimensions2D,
+  PanelOptionsEditorBuilder,
+  StandardEditorContext,
+} from '@grafana/data';
+import { PanelOptionsSupplier } from '@grafana/data/src/panel/PanelPlugin';
+import { NestedValueAccess } from '@grafana/data/src/utils/OptionsUIBuilders';
+import { stylesFactory, usePanelContext, useTheme } from '@grafana/ui';
+import store from 'app/core/store';
+import { GroupState } from 'app/features/canvas/runtime/group';
+import { OptionsPaneCategoryDescriptor } from 'app/features/dashboard/components/PanelEditor/OptionsPaneCategoryDescriptor';
+import { fillOptionsPaneItems } from 'app/features/dashboard/components/PanelEditor/getVisualizationOptions';
+import { setOptionImmutably } from 'app/features/dashboard/components/PanelEditor/utils';
+
+import { DashboardModel } from '../../../features/dashboard/state';
+
+import { InstanceState } from './CanvasPanel';
+import { getElementEditor } from './editor/elementEditor';
+import { getLayerEditor } from './editor/layerEditor';
+
+type Props = {
+  onClose?: () => void;
+  panel: PanelModel;
```

Comment:
why do we need the panel?

Reply:
Right now I'm using the id (not necessary though) and the panel name. I wasn't sure what I'll actually need, that's why I also have the dashboard - which I'm not using it. I'll do a clean up when the PR is ready.